modules/custom/social_graphql/src/GraphQL/EntityConnection.php

```diff
@@ -234,7 +234,8 @@ protected function execute() : SyncPromise {
     // The order is descending if
     // - we want the first results in a reversed query
     // - we want the last results in a non reversed query.
-    $query_order = (!is_null($this->first) && !$this->reverse) || (!is_null($this->last) && $this->reverse) ? 'ASC' : 'DESC';
+    $field_query_order = (!is_null($this->first) && !$this->reverse) || (!is_null($this->last) && $this->reverse) ? $this->queryHelper->getForwardSortDirection() : $this->queryHelper->getReverseSortDirection();
+    $id_query_order = (!is_null($this->first) && !$this->reverse) || (!is_null($this->last) && $this->reverse) ? 'ASC' : 'DESC';
```

Comment:
But these two field has the same result, but for `$field_query_order` uses methods from queryHelper and for the `$id_query_order` uses static values, why not only one variable, this is through https://www.drupal.org/project/social/issues/3191638?

Reply:
We need a fallback sort in case the value of the sorted field is the same. In such cases I think sorting by ID in ascending order (which means "older to newer") makes more sense.

For example in an overview of content sorted by number of likes. With the change in this PR we can decide that sorting by likes means sorting from most to fewest (so descending order) and the reverse sort would be from fewest to most likes (ascending order). In the case of two pieces of content have the same amount of likes I believe it makes more sense to show the oldest content first, rather than have this be decided on how we sort the field.

In the future I think it may make sense to allow choosing the fallback field (for example when sorting by user role we may want to fallback to sorting by name rather than ID) and order but for now just sorting by ID works fine.

In summary: we could use the same value, but we don't have to, so I think we shouldn't.

Reply:
Okay. thank you for the detailed explanation!